core/src/main/java/org/arquillian/cube/impl/client/CubeExtension.java

```diff
@@ -31,17 +32,22 @@ public void register(ExtensionBuilder builder) {
 
         // Arquillian Container integration
         // Only register if container-test-spi is on classpath
-        if(Validate.classExists("org.jboss.arquillian.container.test.spi.client.deployment.AuxiliaryArchiveAppender")) {
+        if (Validate.classExists("org.jboss.arquillian.container.test.spi.client.deployment.AuxiliaryArchiveAppender")) {
             builder.observer(ProtocolMetadataUpdater.class)
                    .observer(CubeContainerLifecycleController.class)
                    .observer(ContainerConfigurationController.class)
                    .observer(CubeRemoteCommandObserver.class);
             builder.service(AuxiliaryArchiveAppender.class, CubeAuxiliaryArchiveAppender.class);
         }
         // Only register if container-test-impl is on classpath
-        if(Validate.classExists("org.jboss.arquillian.container.test.impl.enricher.resource.OperatesOnDeploymentAwareProvider")) {
+        if (Validate.classExists("org.jboss.arquillian.container.test.impl.enricher.resource.OperatesOnDeploymentAwareProvider")) {
             builder.service(ResourceProvider.class, CubeIDResourceProvider.class);
         }
+
+        // Only if recorder-reporter is in classpath we should provide reporting capabilities.
+        if (Validate.classExists("org.arquillian.recorder.reporter.ReporterExtension")) {
+            builder.observer(TakeCubeInformation.class);
+        }
```

Comment:
Yes but it is used all over Arquillian project.https://github.com/arquillian/arquillian-core/blob/c2ae438796b4bab56a86c6246982ecfe038f4a78/core/spi/src/main/java/org/jboss/arquillian/core/spi/LoadableExtension.java#L46 What we can do is add a new method that receives a Class object and inside to the getFullQualifiedName thing an call the String one, so at least we are not breaking anything and in future we can use the Class approach.